bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Environment.java

```diff
@@ -19,131 +19,90 @@
 
 import java.util.Optional;
 
 /**
- * When this class is used as the first argument of an interop method, Ballerina will inject an instance of the class
- * when calling. That instance can be used to communicate with currently executing Ballerina runtime.
+ * When this interface is used as the first argument of an interop method, Ballerina will inject an instance of
+ * the class when calling. That instance can be used to communicate with currently executing Ballerina runtime.
  *
  * @since 2.0.0
  */
-public class Environment {
-
-    private final Strand strand;
-    private Future future;
-    private Module currentModule;
-    private String funcName;
-    private Parameter[] funcPathParams;
-
-    public Environment(Strand strand) {
-        this.strand = strand;
-    }
-
-    public Environment(Strand strand, Module currentModule) {
-        this.strand = strand;
-        this.currentModule = currentModule;
-        future = new Future(this.strand);
-    }
-
-    public Environment(Strand strand, Module currentModule, String funcName, Parameter[] funcPathParams) {
-        this(strand, currentModule);
-        this.funcName = funcName;
-        this.funcPathParams = funcPathParams;
-    }
+public abstract class Environment {
 
     /**
      * Returns the Ballerina function name for the corresponding external interop method.
      *
      * @return function name
      */
-    public String getFunctionName() {
-        return funcName;
-    }
+    public abstract String getFunctionName();
 
     /**
      * Returns an array consisting of the path parameters of the resource function defined as external.
      *
      * @return array of {@link Parameter}
      */
-    public Parameter[] getFunctionPathParameters() {
-        return funcPathParams;
-    }
+    public abstract Parameter[] getFunctionPathParameters();
 
     /**
      * Mark the current executing strand as async. Execution of Ballerina code after the current
-     * interop will stop until given BalFuture is completed. However the java thread will not be blocked
+     * interop will stop until given Ballerina Future is completed. However the java thread will not be blocked
      * and will be reused for running other Ballerina code in the meantime. Therefore callee of this method
-     * must return as soon as possible to avoid starvation of ballerina code execution.
+     * must return as soon as possible to avoid starvation of Ballerina code execution.
      *
-     * @return BalFuture which will resume the current strand when completed.
+     * @return {@link Future} which will resume the current strand when completed.
      */
-    public Future markAsync() {
-        strand.blockedOnExtern = true;
-        strand.setState(State.BLOCK_AND_YIELD);
-        return future;
-    }
+    public abstract Future markAsync();
 
-    public Runtime getRuntime() {
-        return new Runtime(strand.scheduler);
-    }
+    /**
+     * Gets the instance of Ballerina runtime.
```

Comment:
Is the usage of the article `the` correct here? Shouldn't it be `an instance`?

Reply:
Will correct this.

Reply:
Fixed with fb65e40b3e1300337524375cdece55a70d91c234.